gplog/gplog.go

```diff
@@ -286,6 +301,16 @@ func FatalOnError(err error, output ...string) {
 	}
 }
 
+func FatalWithoutPanic(s string, v ...interface{}) {
```

Comment:
I'm not sure it is proper to add this function.
In comments of "abort()", it says
```
 * Abort() is for handling critical errors.  It panic()s to unwind the call stack
 * assuming that the panic is caught by a recover() in the main utility.
```
Does it means we should wrap gplog.Fatal() by ourselves and recover the panic to prevent printing stack trace?

Reply:
However, I would like to have this function in gplog library because it makes our codebase neat.

Reply:
For this PR I recommend add tests for this function and more comments at the top of this file, right after the comments of Info, Verbose, Debug, Warn, Error, Fatal

Reply:
> I'm not sure it is proper to add this function.
> In comments of "abort()", it says
> 
> ```
>  * Abort() is for handling critical errors.  It panic()s to unwind the call stack
>  * assuming that the panic is caught by a recover() in the main utility.
> ```
> 
> Does it means we should wrap gplog.Fatal() by ourselves and recover the panic to prevent printing stack trace?

I need a Fatal function without call stack, because when user see the stack, he may think our program is not robust, like accessing a null pointer. But if we wrap another function and recover the panic, our code would be tedious.

Reply:
> For this PR I recommend add tests for this function and more comments at the top of this file, right after the comments of Info, Verbose, Debug, Warn, Error, Fatal

It is impossible to add test case, for FatalWithoutPanic will exit, then the whole test progress will exit.

Reply:
> It is impossible to add test case, for FatalWithoutPanic will exit, then the whole test progress will exit.

It actually is possible to test `os.Exit` scenarios.  The recommended way is to use `exec.Command` to run the test in another process and check whether it exits (see [here](https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go) for an example of how to do this).

Another option would be adding a function pointer to Exit in the System Functions struct and mock it out with a function that just records that it was called instead of exiting the test run.

Reply:
> > It is impossible to add test case, for FatalWithoutPanic will exit, then the whole test progress will exit.
> 
> It actually is possible to test `os.Exit` scenarios. The recommended way is to use `exec.Command` to run the test in another process and check whether it exits (see [here](https://stackoverflow.com/questions/26225513/how-to-test-os-exit-scenarios-in-go) for an example of how to do this).
> 
> Another option would be adding a function pointer to Exit in the System Functions struct and mock it out with a function that just records that it was called instead of exiting the test run.

I mock the os.Exit by monkey.